client/session/profile_manager_test.go

```diff
@@ -0,0 +1,341 @@
+package session_test
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"testing"
+
+	"github.com/franela/goblin"
+	. "github.com/onsi/gomega"
+	"github.com/spacelift-io/spacectl/client/session"
+)
+
+func TestProfileManager(t *testing.T) {
+	g := goblin.Goblin(t)
+	RegisterFailHandler(func(m string, _ ...int) { g.Fail(m) })
+
+	g.Describe("ProfileManager", func() {
+		var testDirectory string
+		var profilesDirectory string
+		var manager *session.ProfileManager
+
+		g.BeforeEach(func() {
+			var err error
+			if testDirectory, err = ioutil.TempDir("", "spacectlProfiles"); err != nil {
+				t.Errorf("Could not create a temp profiles directory: %w", err)
+			}
+
+			profilesDirectory = path.Join(testDirectory, "profiles")
+
+			manager = session.NewProfileManager(profilesDirectory)
+		})
+
+		g.AfterEach(func() {
+			if err := os.RemoveAll(testDirectory); err != nil {
+				g.Fail(fmt.Errorf("Failed to cleanup temp profiles directory: %w", err))
```

Comment:
I'm not 100% sure you can fail in before/after hooks in Goblin. I used `testify/require` in those cases.

Reply:
@marcinwyszynski I've just tried it out there, and it looks like using `Fail()` works, but using `require` causes the test run to hang.

For example, I tried adding the following to a `BeforeEach()`:

```go
require.Nil(t, "abc123", "Testing")
```

If I do that, the test execution just seems to pause at that point. But if I do the following it fails the tests correctly:

```go
g.Fail("testing 123")
```

Reply:
OK, my bad then.